pkg/util/kubernetes/cloudprovider/cloudprovider_dca.go

```diff
@@ -0,0 +1,60 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
```

Comment:
The kubeapiserver flag is also used for the agent.
Do we want to limit this somehow? Otherwise, it means that code from the agent, for example, KSM when configured to run on the node agent will be able to call this function and make request to the API server.

Reply:
What are the ways to limit it ?

Reply:
The build tags are defined in `tasks/build_tags.py`.

As you can see `kubeapiserver` appears both in `AGENT_TAGS` and `CLUSTER_AGENT_TAGS`. 

There isn't a specific "cluster-agent" tag. I think for code that's specific for the cluster-agent and it's about clusterchecks we use `clusterchecks`. For others things we can use a combination like `//go:build kubeapiserver && !kubelet` (for example: `pkg/util/kubernetes/hostname.go`) that we know is specific for the DCA. Although I agree this is not very clean.

Reply:
Fixed